ydb/core/mind/bscontroller/commit_config.cpp

```diff
@@ -0,0 +1,80 @@
+#include "impl.h"
+#include <ydb/library/yaml_config/yaml_config_parser.h>
+#include <ydb/core/blobstorage/nodewarden/node_warden_impl.h>
+#include <ydb/core/blobstorage/nodewarden/distconf.h>
+
+namespace NKikimr::NBsController {
+
+    class TBlobStorageController::TTxCommitConfig
+        : public TTransactionBase<TBlobStorageController>
+    {
+        TString Config;
+        ui32 ConfigVersion;
+        NKikimrBlobStorage::TStorageConfig StorageConfig;
+        bool Success = true;
+        std::optional<TString> ErrorReason;
+        TConsoleInteraction::TCommitConfigResult Result;
+        bool SkipBSCValidation;
+
+    public:
+        TTxCommitConfig(TBlobStorageController *controller, const TString& config, ui32 configVersion, bool skipBSCValidation)
+            : TTransactionBase(controller)
+            , Config(config)
+            , ConfigVersion(configVersion)
+            , SkipBSCValidation(skipBSCValidation)
+        {}
+
+        TTxType GetTxType() const override { return NBlobStorageController::TXTYPE_COMMIT_CONFIG; }
+    
+        bool ParseConfig(TString& config, ui32& configVersion) {
+            NKikimrConfig::TAppConfig appConfig;
+            try {
+                appConfig = NKikimr::NYaml::Parse(config);
+            } catch (const std::exception& ex) {
+                Result.Status = TConsoleInteraction::TCommitConfigResult::EStatus::ParseError;
+                return false;
+            }
+            const bool success = NKikimr::NStorage::DeriveStorageConfig(appConfig, &StorageConfig, nullptr);
+            return success;
+        }
+
+        bool Execute(TTransactionContext& txc, const TActorContext&) override {
+            NIceDb::TNiceDb db(txc.DB);
+            if (!ParseConfig(Config, ConfigVersion)) {
+                Success = false;
+                Result.Status = TConsoleInteraction::TCommitConfigResult::EStatus::ParseError;
+                return true;
+            }
+            if (!SkipBSCValidation) {
+                auto ErrorReason = NKikimr::NStorage::ValidateConfig(StorageConfig);
+                if (!ErrorReason) {
+                    Success = false;
+                    Result.Status = TConsoleInteraction::TCommitConfigResult::EStatus::ValidationError;
+                    return true;
+                }
+            }
+            db.Table<Schema::State>().Key(true).Update(
+                NIceDb::TUpdate<Schema::State::YamlConfig>(Config),
+                NIceDb::TUpdate<Schema::State::ConfigVersion>(ConfigVersion));
+            return true;
+        }
+
+        void Complete(const TActorContext& ctx) override {
+            if (Success) {
+                Self->StorageConfig = StorageConfig;
```

Comment:
I'd prefer some checks against races (what if Self->StorageConfig has changed since the whole operation has been started?)